dashboard/app/config.go

```diff
@@ -690,34 +708,35 @@ func (cfg *Config) lastActiveReporting() int {
 	return last
 }
 
-var kernelReposKey = "Custom list of kernel repositories"
-
-func contextWithRepos(c context.Context, list []KernelRepo) context.Context {
-	return context.WithValue(c, &kernelReposKey, list)
-}
-
-func getKernelRepos(c context.Context, ns string) []KernelRepo {
-	if val, ok := c.Value(&kernelReposKey).([]KernelRepo); ok {
-		return val
-	}
-	return config.Namespaces[ns].Repos
+func isDecommissioned(c context.Context, ns string) bool {
+	return getConfig(c).Namespaces[ns].Decommissioned
 }
 
-var decommKey = "Custom decommissioned status"
-
-func contextWithDecommission(c context.Context, ns string, value bool) context.Context {
-	mm, _ := c.Value(&decommKey).(map[string]bool)
-	if mm == nil {
-		mm = map[string]bool{}
+// Create a shallow copy of GlobalConfig with a replaced namespace config.
+func replaceNamespaceConfig(c context.Context, ns string, f func(*Config) *Config) *GlobalConfig {
```

Comment:
replaceNamespaceConfig/replaceManagerConfig are intended only for tests, and used only in tests, so move to util_test.go to make it clear.

Also can make sense to fold contextWithConfig into replaceNamespaceConfig, it's the only caller. replaceNamespaceConfig can return new context.

Reply:
> replaceNamespaceConfig/replaceManagerConfig are intended only for tests, and used only in tests, so move to util_test.go to make it clear.

Done

> Also can make sense to fold contextWithConfig into replaceNamespaceConfig, it's the only caller. replaceNamespaceConfig can return new context.

I think it's not really necessary. These methods are doing different things and they look cleaner as is.